### PR TITLE
feat(builtin-groups): support builtin groups in user.yaml

### PR DESCRIPTION
**Pull request by rudyardrichter**
### New Features

- Allow setting policies for `anonymous` and `logged-in` groups in the `user.yaml`.

### CODE REVIEW

**Comment by PlanXCyborg** [comment]
This PR contains code that is not formatted correctly according to [`black`](https://github.com/ambv/black). Run `black` on your code before merging.

<details>
<summary>Expand the full diff to see formatting changes</summary>

```diff
--- fence/rbac/client.py
+++ blackened
@@ -480,11 +480,13 @@
                 "could not grant policy `{}` to group `{}`: {}".format(
                     policy_id, group_name, msg
                 )
             )
             return None
-        self.logger.info("granted policy `{}` to group `{}`".format(policy_id, group_name))
+        self.logger.info(
+            "granted policy `{}` to group `{}`".format(policy_id, group_name)
+        )
         return data
 
     @_arborist_retry()
     def create_user_if_not_exist(self, username):
         self.logger.info("making sure user exists: `{}`".format(username))
```
</details></br>

This formatting comment was generated automatically by a script in [uc-cdis/wool](https://github.com/uc-cdis/wool).

File: fence/rbac/client.py [hunk 1]
```diff
@@ -466,6 +466,25 @@ def create_group(
...
 
+    @_arborist_retry()
+    def grant_group_policy(self, group_name, policy_id):
+        url = self._group_url + "/{}/policy".format(group_name)
+        request = {"policy": policy_id}
+        response = requests.post(url, json=request)
+        data = _request_get_json(response)
+        if response.status_code != 204:
+            msg = data.get("error", "unhelpful response from arborist")
+            if isinstance("error", dict):
```
Comment (Avantol13) [inline fence/rbac/client.py]:
```suggestion
            if isinstance(data, dict):
```

File: fence/rbac/client.py [hunk 2]
```diff
@@ -466,6 +466,25 @@ def create_group(
...
+                msg = data["error"].get("message", msg)
```
Comment (Avantol13) [inline fence/rbac/client.py]: do you need to check `if "error" in data`, or do non-204s always have that information?

Comment (rudyardrichter) [reply fence/rbac/client.py]: should always be there but never hurts :+1:

**Pull request merged**